src/Nuvemshop/Provider.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace SocialiteProviders\Nuvemshop;
+
+use GuzzleHttp\Exception\GuzzleException;
+use GuzzleHttp\RequestOptions;
+use SocialiteProviders\Manager\OAuth2\AbstractProvider;
+use SocialiteProviders\Manager\OAuth2\User;
+
+/**
+ * @see https://tiendanube.github.io/api-documentation/authentication
+ */
+class Provider extends AbstractProvider
+{
+    public const IDENTIFIER = 'NUVEMSHOP';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $scopes = [
+        'read_content',
+        'write_content',
+        'read_products',
+        'write_products',
+        'read_customers',
+        'write_customers',
+        'read_orders',
+        'write_orders',
+        'read_coupons',
+        'write_coupons',
+        'write_scripts',
+        'write_shipping'
+    ];
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $scopeSeparator = ' ';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getAuthUrl($state)
+    {
+        return $this->buildAuthUrlFromBase(
```

Comment:
`return $this->buildAuthUrlFromBase("https://www.tiendanube.com/apps/{$this->clientId}/authorize", $state);`

is just fine